client/components/command-palette/index.tsx

```diff
@@ -242,31 +237,28 @@ const CommandPalette = () => {
 	const [ isOpen, setIsOpen ] = useState( false );
 	const [ footerMessage, setFooterMessage ] = useState( '' );
 	const [ emptyListNotice, setEmptyListNotice ] = useState( '' );
-	const currentRoute = useSelector( ( state: object ) => getCurrentRoutePattern( state ) );
-	const dispatch = useDispatch();
+	// TODO: Find an alternative way to use the current route.
+	//const currentRoute = useSelector( ( state: object ) => getCurrentRoutePattern( state ) );
+	const currentRoute = null;
 	const open = useCallback( () => {
 		setIsOpen( true );
-		dispatch(
-			recordTracksEvent( 'calypso_hosting_command_palette_open', {
-				current_route: currentRoute,
-			} )
-		);
-	}, [ dispatch, currentRoute ] );
+		recordTracksEvent( 'calypso_hosting_command_palette_open', {
+			current_route: currentRoute,
+		} );
+	}, [ currentRoute ] );
 	const close = useCallback< CommandMenuGroupProps[ 'close' ] >(
 		( commandName = '', isExecuted = false ) => {
-			dispatch(
-				recordTracksEvent( 'calypso_hosting_command_palette_close', {
-					// For nested commands the command.name would be the siteId
-					// For root commands the selectedCommandName would be empty
-					command: selectedCommandName || commandName,
-					current_route: currentRoute,
-					search_text: search,
-					is_executed: isExecuted,
-				} )
-			);
+			recordTracksEvent( 'calypso_hosting_command_palette_close', {
+				// For nested commands the command.name would be the siteId
+				// For root commands the selectedCommandName would be empty
+				command: selectedCommandName || commandName,
+				current_route: currentRoute,
+				search_text: search,
+				is_executed: isExecuted,
+			} );
```

Comment:
I'm wondering if it would make more sense to decouple tracking from the palette. We could dispatch a tracking event and let the underlying system handle if and how to track the palette events. A client implementing the palette might not be interested in these specific events or might not have access to them.

Reply:
> A client implementing the palette might not be interested in these specific events or might not have access to them.

We're implementing all of the clients, we share the same interest in these events across all clients. i.e. an event is useful across all clients or useless across all clients.

Reply:
Agreed with Dean. We want to record the same events in Calypso and WP Admin.